### DISCUSSION TITLE
Для не админа показывается "A PHP Error was encountered"

### DISCUSSION THREAD

**Discussion by kelemator** [maxsite/cms#481] [Q&A] [answered]
Здравствуйте.

На морде сайта для всех страниц там, где обычно иконка "edit page" для админа, а у остальных "ничего", вылезает такая ошибка:

```
A PHP Error was encountered
Severity: 8192
Message: str_replace(): Passing null to parameter #2 ($replace) of type array|string is deprecated
Filename: core/page-out.php
Line Number: 513
```

И в подвале тогда показывается примерно такое, без цифр "Время: {elapsed_time} | Память: {memory_usage} ".

Если залогиниться, то ошибка исчезает.

В самом файле "ядра" строчка 513: $out = str_replace('[edit]', $edit, $out);

Не пойму, что с этим можно сделать. Обновил автоматически движок до последней версии, но ошибка осталась. 

На сервере - php 8.1

Буду благодарен за помощь :)

**Comment by maxsite** [comment]
А на дефолтном шаблоне тоже ошибка?

**Comment by kelemator** [comment]
да, у меня дефолтный шаблон. сейчас взял полностью его стер и вручную заменил на файлы из релиза - ошибка осталась и сайт еще ко всему поменялся (похоже, это из-за того, что компоненты поменяли название на header6, header7, header11), но это наверное я как-то починю.
а вот ошибка все равно выводится.. и в админке в подвале тоже:

```
A PHP Error was encountered
Severity: 8192
Message: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated
Filename: core/Output.php
Line Number: 367
```

может на сервере поменять версию php на более старую? сейчас там 8.1

**Comment by maxsite** [comment, marked as answer]
Попробуйте поменять на PHP 8.0. Я на 8.1 ещё не тестировал — почему-то версия не хочет у меня работать на компе. Возможно в этом дело.

**Comment by kelemator** [comment]
Поменял на 8.0 - заработало на обновленном шаблоне. Спасибо! :)

**Comment by maxsite** [comment]
Хорошо, спасибо. Буду думать как бы потестировать на новой версии PHP...

**Comment by maxsite** [comment]
Внес изменения в CodeIgniter для PHP 8.1 https://github.com/maxsite/cms/commit/c2c0e976d49a3379a23a050a2a426528c419c8b6 Будет время, посмотрите.

**Reply by smogendrr** [reply]
У меня та же проблема была, патч к CodeIgniter помог в случае с Filename: core/Output.php,
А вот с Filename: core/page-out.php проблема осталась, туда также добавил по примеру выше (string) к аргументам функции, и теперь норм

пс: php8.1
